application/lib/view/widgets/custom_search_bar.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:flutter/material.dart';
+import 'package:application/utils/colors.dart';
+import 'package:application/utils/translation_en.dart';
+import 'package:application/view_model/products_screen_view_model.dart';
+
+class CustomSearchBar extends StatelessWidget {
+  final TextEditingController controller;
+  final ProductsScreenViewModel viewModel;
```

Comment:
Egy ViewModel egy View-hoz tartozik. Azért nem jó taktika, ha megosztod a viewmodeleket a view-k között, mert nem lesz egyértelmű, hogy éppen melyik view hívta a változtatásokat és mi miért történt. 
Ha arra van szükséged, hogy egy Widget egy külső viewmodelben lévő változót tudjon módosítani, mint itt a searchTerm-et, akkor erre van egy jobb taktika. 
Az onChanged függvényt várja a CustomSearchBar bemenő paraméterként. Szóval legyen egy adattagja a CustomSearchBar-nak: 
`final VoidCallback updateFunction; `
 Amikor pedig a CustomSearchBar-t beágyazod a product screenen, akkor 
`CustomSearchBar(updateFunction: viewModel.updateSearchTerm)` 
() nélkül, mert nem meghívni akarjuk az updateSearchTerm függvényt, hanem átadni.
A product screenen pedig hozzáférsz a viewModel-hez úgyhogy a viewmodelben legyen egy függvény:
`void updateSerachTerm(String value) {
        searchTerm = value;
      }`